compiler/src/iree/compiler/Codegen/Common/LowerUKernelDescriptors.cpp

```diff
@@ -113,11 +108,19 @@ static LogicalResult convertToUKernelGeneric(RewriterBase &rewriter,
       }
     }
   }
+
   rewriter.setInsertionPoint(op);
-  rewriter.replaceOpWithNewOp<IREE::Codegen::UKernelGenericOp>(
-      op, op->getResults().getTypes(), name, tensorInputs, tensorOutputs,
-      otherOperands, DictionaryAttr(),
-      /*strided_outer_dims=*/0);
+  if (!provider) {
+    // Default ukernel generic op is created when a provider doesn't exist.
+    rewriter.replaceOpWithNewOp<IREE::Codegen::UKernelGenericOp>(
+        op, op->getResults().getTypes(), name, tensorInputs, tensorOutputs,
+        otherOperands, DictionaryAttr(),
+        /*strided_outer_dims=*/0);
+  } else if (failed(provider.createAndReplaceWithUkernelOp(
+                 rewriter, name, targetConfiguration, op, tensorInputs,
+                 tensorOutputs, otherOperands))) {
+    return failure();
+  }
   return success();
```

Comment:
nit: I think it's more natural to write this as:

```suggestion
  if (!provider) {
    // Default ukernel generic op is created when a provider doesn't exist.
    rewriter.replaceOpWithNewOp<IREE::Codegen::UKernelGenericOp>(
        op, op->getResults().getTypes(), name, tensorInputs, tensorOutputs,
        otherOperands, DictionaryAttr(),
        /*strided_outer_dims=*/0);
      return success();
  }
  return provider.createAndReplaceWithUkernelOp(
                 rewriter, name, targetConfiguration, op, tensorInputs,
                 tensorOutputs, otherOperands)));
```